### PR TITLE
Remove eager loding of shipping_methods and shipping_rates from customer_totals so that report runs faster

### LINKED ISSUE #4547: Timeout on Orders and Fulfilment Report
## Description

This report is timing out in UK and Australia.... detail

## Expected Behavior

Report loads in 3-4 minutes

## Actual Behaviour

Timeout at 12 minutes in UK

## Steps to Reproduce

1. UK Timeout:
Customer Totals Report
A user with only access to one enterprise - lynnedavis[at]gmail[dot]com
Tamar Valley Food Hub
Dates: 2019-11-29 to 2019-12-05
OC: TVFH20191206

## Workaround

To run the report 1 or 2 days at a time.

## Severity

bug-s2: a non-critical feature is broken, no workaround

## Your Environment

* Version used: 2.6.6
* Browser name and version: Chrome Firefox
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #4547

These includes make the main query much slower.
Without these includes I see 278 queries instead of 209 queries but the report loads in 50secs instead of 4.48mins.

These are my local tests with different includes in CustomerTotals#line_item_includes:
```
        # original version: https://github.com/openfoodfoundation/openfoodnetwork/commit/8a57977b448937b81a2f058546183e8d269d707a
        [{ variant: :product,
           order: [:bill_address, :shipments, :order_cycle, :adjustments] }] # 42secs - 649 queries

        [{ variant: :product,
           order: [:bill_address, :ship_address, :order_cycle, :adjustments, :payments,
                                        :shipments] }] # 42s - 564 queries

        [{ variant: :product,
           order: [:bill_address, :ship_address, :order_cycle, :adjustments, :payments,
                   :user, :distributor, :shipments] }] # 44s 516 queries

        [{ variant: [{ option_values: :option_type }, { product: :supplier }],
           order: [:bill_address, :ship_address, :order_cycle, :adjustments, :payments,
                   :user, :distributor, :shipments] }] # 50 secs 278 queries

        [{ variant: :product,
           order: [:bill_address, :ship_address, :order_cycle, :adjustments, :payments,
                   shipments: { shipping_rates: :shipping_method }] }] # 1,92mins - 495 queries

        # LIVE version
        [{ variant: [{ option_values: :option_type }, { product: :supplier }],
           order: [:bill_address, :ship_address, :order_cycle, :adjustments, :payments,
                   :user, :distributor, shipments: { shipping_rates: :shipping_method }] }] # 4.48mins - 209 queries
``` 

#### What should we test?
Run the report with live data and see that is runs much faster without timing out.
This wont break the report for sure so we may just push this to prod and see the impact.

#### Release notes
Changelog Category: Changed
Improved performance of Orders and Fullfilment report Customer Totals by improving the database query.

### CODE REVIEW

**Comment by luisramos0** [comment]
I spent a full day replacing the very long IN clause from the query (100k order ids and 70k line items ids) in #4549 just to realize that the main problem was in the includes...
The report with this PR and 4549 runs in 42sec (275 queries) instead of 50secs (279 queries), so 4549 will also improve performance.

**Comment by sauloperez** [comment]
I don't really understand how this improves the situation. Two questions come to my mind:

* If we need to remove these, did we introduce a regression at some point? Adding includes is something we just did recently.
* If the data from these tables is needed in the report, does this mean we now introduce N+1? How can that be faster?

**Comment by luisramos0** [comment]
Adding includes can mess up the execution plan of a query (imagine for example joining all orders in the DB with their associated tables before filtering the list of orders to the orders in the OC). So in this case removing an include makes the query go faster in the DB and then we load the data in N+1 queries, in this case, instead of 209 we make 278 queries. We run 69 more queries but the initial very complex query runs in a few seconds instead of 5 minutes.

**Comment by luisramos0** [comment]
I just confirmed the load times locally, with uk live data and the exact same criteria as the S2 bug, I get 5mins before this PR and 52secs after this PR.

File: lib/open_food_network/orders_and_fulfillments_report/customer_totals_report.rb [hunk 1]
```diff
@@ -192,7 +192,7 @@ def columns
       def line_item_includes
         [{ variant: [{ option_values: :option_type }, { product: :supplier }],
            order: [:bill_address, :ship_address, :order_cycle, :adjustments, :payments,
-                   :user, :distributor, shipments: { shipping_rates: :shipping_method }] }]
```
Comment (Matt-Yorkley) [inline lib/open_food_network/orders_and_fulfillments_report/customer_totals_report.rb]: If we're removing these two includes, we can remove the private method below as well.

Comment (luisramos0) [reply lib/open_food_network/orders_and_fulfillments_report/customer_totals_report.rb]: yeah, I would agree but, I am not sure exactly why, removing it increases the load time on my test in around 10 secs

**Comment by kristinalim** [comment]
Thanks for your tip a couple of weeks ago @luisramos0 about swap just not being freed. About current used swap in UK, I ran `vmstat 1` and noting that the used swap seems to be inactive:

```
procs -----------memory---------- ---swap-- -----io---- -system-- ------cpu-----
 r  b   swpd   free   buff  cache   si   so    bi    bo   in   cs us sy id wa st
 1  0 549580 1602588 312908 1649996    0    0     0    72  166  322  0  0 100  0  0
 0  0 549580 1602588 312908 1649996    0    0     0    52  138  273  0  0 100  0  0
 0  0 549580 1602652 312908 1649996    0    0     0     4  613 1199  2  0 98  0  0
```

**Comment by kristinalim** [comment]
Moving this to Test Ready because of the two approvals and no request for revision. @sauloperez If you think we should not merge this, do feel free to move this back to Code Review.

**Comment by luisramos0** [comment]
nice @kristinalim 
I wonder how do you see that the used swap seems to be inactive. I know this is a common thing from some past experience but I cant remember any details :rofl: 

Anyway I will test this afternoon again and see if when running the report in master I see the swap being used. Could be it.

I am running with uk live data, all the DB without any anonymization or truncation.

**Comment by lin-d-hop** [comment]
I gave this a gander on all O&F reports.

Everything looks fine.

I seem to be locked out of all accounts bulk data for testing.... On UK and FR staging. So no tests on performance have been done at all. That can happen on production.

Good to go :-)

**Pull request merged**